modules/user/user.repository.js

```diff
@@ -45,44 +66,36 @@ class UserRepository {
     }
 
     async saveNewUser(newUser) {
-        let usersArr = await this.readAllUsers();
-        usersArr.push(newUser);
+        const newMongoUser = new UserModel(newUser);
+        await newMongoUser.save();
+        const savedUser = await UserModel.findOne(newMongoUser);
 
-        await this.saveFileUsers(usersArr);
+        if (!savedUser) {
+            throw new Error(500);
+        }
 
-        return [newUser];
+        return [savedUser];
     }
 
-    async updateUser(updatedUser) {
-        const usersArr = await this.readAllUsers();
-        const userIndex = await this.findUserIndexById(updatedUser.id);
-
-        usersArr[userIndex] = updatedUser;
-
-        await this.saveFileUsers(usersArr);
+    async updateUser(updatedUserData) {
+        mongoose.set("useFindAndModify", false);
```

Comment:
I am not sure, that this code relates to update user logic. This looks like mongoose configuration, are you sure that this code should be here?